README.md

```diff
@@ -37,6 +37,18 @@ For scalability, the code uses [MPI Neighbourhood collectives](https://www.mpi-f
 - Reading and writing meshtags associated to meshes `adios4dolfinx.read/write_meshtags`
 - Reading checkpoints for any element (serial and parallel, one checkpoint per file). Use `adios4dolfinx.read/write_function`.
 
+> [!IMPORTANT]  
+> For a checkpoint to be valid, you first have to store the mesh with `write_mesh`, then use `write_function` to append to the checkpoint file.
```

Comment:
I had missed that, I'll fix my I/O. I guess my RBniCSx test didn't see that because I am using builtin meshes, so even if I was not reading them in I was assured (except for changes in upstream dolfinx) that the mesh was the same anyways?

Reply:
However, what will happen if the user tries to call `read_function` passing as argument a `dolfinx.fem.Function` defined on a mesh that is not the one obtained with `read_mesh` on the same file?

Reply:
To give you more context, in my use cases I would have the user read the mesh from a mesh checkpoint file, and then read time dependent solutions corresponding to (say) 100 different solves of a time dependent problem on 150 time steps. The 100 different solves are obtained e.g. for different coefficients in the PDE. 

Each of the 100 problems, and each of the 150 time instances of the same problem, share the same mesh. The time history associated to solutions of problem 1 will be stored in checkpoint file 1 with the new time dependency feature. However, solutions of problem 2 will be stored in checkpoint file 2, and not appended at the end of checkpoint file 1, because they don't belong together (time would jump back to zero at the beginning of problem 2).

Do I need to have 100 copies of the mesh (i.e., one for every checkpoint file)? This `IMPORTANT` block seems to say so, but since we started this discussion I'd rather double check.

Reply:
With the new features, you could store all the checkpoints in the same file, as they have their own time dependency array.

I cannot support reading the mesh from another file, as the ordering of cells might differ if they are in another input file (and also the very crucial cell permutations for vector-spaces).